python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -854,82 +854,64 @@ def get_parent_asset_partitions_updated_after_child(
             )
         return updated_parents
 
-    def get_outdated_ancestors(
-        self, *, asset_partition: AssetKeyPartitionKey
-    ) -> AbstractSet[AssetKey]:
-        """Return the set of assets that are ancestors of the given asset partition and have parents
-        that have been updated more recently than they have.
+    def have_ignorable_partition_mapping_for_outdated_check(
```

Comment:
Definitely a weird spot for it, but I would say that `get_outdated_parents` is also "a method that's aware of the existence of a particular rule". `get_outdated_parents` does sorta need to live on the CachingInstanceQueryer so that the results can be cached, so this method also fits best here (as it means we don't have to pass in a function into `get_outdated_parents`). I think a rename here and an update of the comment would probably help make this feel a bit less bad.

Reply:
Maybe `have_ignorable_partition_mapping_for_outdated_computation`?

Reply:
Bringing the conversation back here from Slack: it seems like directionally we want to get to a point where "outdated" or whatever we call it is a Dagster-wide concept that matches the "Upstream data" label we show in the UI. So that would mean that we'd want to harmonize the ignorable partition mapping criteria across those two different settings.  And means that this ends up as a general concept that's not tied to a particular rule.

So yes I think that name is fine. I think could maybe get rid of "_computation". Also I think "has" is more idiomatic that "have" here?